src/frontend/admin/src/components/templates/enrollments/buttons/EnrollmentActions.tsx

```diff
@@ -0,0 +1,163 @@
+  },
+});
+
+type Props = {
+  enrollment: Enrollment;
+};
+
+export default function EnrollmentActionsButton({ enrollment }: Props) {
+  const intl = useIntl();
+  const validateUnrollModal = useModal();
+  const enrollmentsQuery = useEnrollments({}, { enabled: false });
+  const enrollmentQuery = useEnrollment(enrollment.id);
+  const canReactivate =
+    moment(enrollment.course_run.enrollment_end).diff(moment()) > 0;
+
+  const updateEnrollment = async (is_active: boolean) => {
+    await enrollmentQuery.methods.update(
+      { id: enrollment.id, is_active },
+      {
+        onSuccess: enrollmentsQuery.methods.invalidate,
+      },
+    );
+  };
+
+  const handleEnrollUnroll = async () => {
+    if (enrollment.is_active) {
+      validateUnrollModal.handleOpen();
+      return;
+    }
+
+    await updateEnrollment(!enrollment.is_active);
+  };
+
+  const options = useMemo(() => {
+    const allOptions: MenuOption[] = [];
+    const isActive = enrollment.is_active;
+    allOptions.push({
+      icon: isActive ? <CancelIcon /> : <CheckCircleIcon />,
+      isDisable: isActive ? false : !canReactivate,
+      disableMessage: isActive
+        ? undefined
+        : intl.formatMessage(messages.cantEnrollMessage),
+      mainLabel: intl.formatMessage(
+        isActive ? messages.unroll : messages.enroll,
+      ),
+      onClick: handleEnrollUnroll,
+    });
+    return allOptions;
+  }, [enrollment]);
+
+  if (options.length === 0) {
+    return undefined;
+  }
+
+  return (
+    <>
+      <ButtonMenu
+        data-testid="enrollment-view-action-button"
+        label={intl.formatMessage(messages.enrollmentActionsLabel)}
+        id="order-view-action-button"
+        variant="contained"
+        color="secondary"
+        options={options}
+      />
+      <AlertModal
+        validateColorButton="error"
+        validateLabel={intl.formatMessage(messages.unroll)}
+        title={intl.formatMessage(messages.validateUnrollModalTitle)}
+        fullWidth
+        maxWidth="sm"
+        handleAccept={() => updateEnrollment(!enrollment.is_active)}
+        message={intl.formatMessage(
+          canReactivate
+            ? messages.validateUnrollModalMessage
+            : messages.validateUnrollButImpossibleToEnrollAfterModalMessage,
+          {
+            br: (
+              <>
+                <br />
+                <br />
+              </>
+            ),
+            username: (
+              <strong>
+                {enrollment.user.full_name && enrollment.user.full_name !== ""
+                  ? enrollment.user.full_name
+                  : enrollment.user.username}
+              </strong>
+            ),
+            courseRunTitle: <strong>{enrollment.course_run.title}</strong>,
+            enrollmentEndDate: (
+              <strong>
+                {new Date(
+                  enrollment.course_run.enrollment_end!,
+                ).toLocaleString()}
```

Comment:
In order to improve readability, I'll do not display seconds then display month in full text. (e.g: 4 April 2024 06:31)

Maybe this is out of the scope of this PR as this suggestion could be applied to the whole application.